### PR TITLE
test: Break down huge monolith mock header to improve test compilation performance

### PR DESCRIPTION
**Pull request by foreseeable**
Commit Message: breakdown `test/mocks/server/mocks.h` into different mock classes
Additional Description:  `test/mocks/server/mocks.h` is a wide-used mock header included by various test files. However it's very huge and most test files only used a small portion of it. Splitting it up into different mock classes will be helpful to reduce compilation time. 
Risk Level: low
Testing: existing tests
Docs Changes: N/A
Release Notes: no
Related Issues: #10917

### CODE REVIEW

**Comment by foreseeable** [comment]
@ahedberg

File: test/mocks/server/drain_manager.cc [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#include "drain_manager.h"
+
+#include <string>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::_;
```
Comment (ahedberg) [inline test/mocks/server/drain_manager.cc:8]: As I mentioned on the last PR, these should be fully-qualified and inside the innermost working namespace, or removed if not being used.

File: test/mocks/server/bootstrap_extension_factory.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  MockBootstrapExtensionFactory();
+  ~MockBootstrapExtensionFactory() override;
+
+  MOCK_METHOD(BootstrapExtensionPtr, createBootstrapExtension,
+              (const Protobuf::Message&, Configuration::ServerFactoryContext&), (override));
+  MOCK_METHOD(ProtobufTypes::MessagePtr, createEmptyConfigProto, (), (override));
+  MOCK_METHOD(std::string, name, (), (const, override));
+};
+} // namespace Configuration
+
```
Comment (ahedberg) [inline test/mocks/server/bootstrap_extension_factory.h:26]: Removing the blank lines between closing namespace brackets would be more consistent with the opening namespace brackets.

File: test/mocks/server/drain_manager.cc (hunk 1 above)
Comment (foreseeable) [reply test/mocks/server/drain_manager.cc]: Thanks for let me know, now they should be fully qualified and stay inside the innermost namespace.

File: test/mocks/server/bootstrap_extension_factory.h (hunk 2 above)
Comment (foreseeable) [reply test/mocks/server/bootstrap_extension_factory.h]: Now they're removed

File: test/mocks/server/admin_stream.h [hunk 3]
```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <chrono>
```
Comment (ahedberg) [inline test/mocks/server/admin_stream.h:3]: Are we scripting the removal of extra #includes or doing it manually? We don't need most of these system headers in the new headers.

File: test/mocks/server/mocks.cc [hunk 4]
```diff
@@ -1,295 +1,3 @@
...
-}
-
-MockTracerFactoryContext::~MockTracerFactoryContext() = default;
-
-MockBootstrapExtensionFactory::MockBootstrapExtensionFactory() = default;
-MockBootstrapExtensionFactory::~MockBootstrapExtensionFactory() = default;
-} // namespace Configuration
-} // namespace Server
-} // namespace Envoy
+namespace Envoy {}
```
Comment (ahedberg) [inline test/mocks/server/mocks.cc:3]: We can delete the .cc file.

File: test/mocks/server/mocks.h [hunk 5]
```diff
@@ -1,661 +1,27 @@
...
+#include "overload_manager.h"
+#include "server_lifecycle_notifier.h"
+#include "tracer_factory.h"
+#include "tracer_factory_context.h"
+#include "transport_socket_factory_context.h"
+#include "watch_dog.h"
+#include "worker.h"
+#include "worker_factory.h"
+
+namespace Envoy {}
```
Comment (ahedberg) [inline test/mocks/server/mocks.h:27]: This empty namespace block can be removed.

File: test/mocks/server/admin_stream.h (hunk 3 above)
Comment (foreseeable) [reply test/mocks/server/admin_stream.h:3]: I remove them manually, just forget to remove them. Thanks for pointing out

**Comment by foreseeable** [comment]
cc @asd

> Please fix formatting. `./ci/run_envoy_docker.sh './ci/do_ci.sh fix_format'` should do it.

Now it should be fixed. Actually the empty namespace block in mocks.h from previous commits  was used to make the formatter happy. now I found I can use no lint instead

**Pull request merged**